tests/functional/test_macro_extract.js

```diff
@@ -0,0 +1,42 @@
+import { expect } from 'chai';
+import * as babel from '@babel/core';
+import fs from 'fs';
+import ttag from 'src/plugin';
+import { rmDirSync } from 'src/utils';
+import dedent from 'dedent';
+
+const output = 'debug/translations.pot';
+const options = {
+    plugins: [[ttag, { extract: { output } }]],
+};
+
+const expect2 = `msgid ""
+msgstr ""
+"Content-Type: text/plain; charset=utf-8\\n"
+"Plural-Forms: nplurals=2; plural=(n!=1);\\n"
+
+msgid "test"
+msgstr ""
+
+msgid "test2"
+msgstr ""`;
+
+describe('Macro extract', () => {
+    before(() => {
+        rmDirSync('debug');
+    });
+
+    it('should extract translations from macro', () => {
+        const input = dedent(`
+            import { t, jt } from 'babel-plugin-ttag/dist/ttag.macro';
```

Comment:
Since the macro script need to import main plugin and some constants from `defaults` file, so the macro script should be placed in `src/` folder and exported into `dist/` folder.

@AlexMost @MrOrz How yours thought for this entry?

Reply:
I guess `babel-plugin-ttag/dist/ttag.macro` is too verbose and it makes sense to create a separate package `ttag.macro` that will just reexport `babel-plugin-ttag/dist/ttag.macro`, so we can leave it inside the `dist` folder for now.

Reply:
So the `ttag.macro` is another version of `babel-plugin-ttag` package?
Just change the entry from `dist/plugin.js` to `dist/ttag.macro.js`?

Reply:
As i can see it can install 'babel-plugin-ttag' as a dep and reexport `babel-plugin-ttag/dist/ttag.macro.js`;